lib/Doctrine/Common/Annotations/Reader.php

```diff
@@ -26,9 +26,11 @@ public function getClassAnnotations(ReflectionClass $class);
      *
      * @param ReflectionClass $class          The ReflectionClass of the class from which
      *          the class annotations should be read.
-     * @param string          $annotationName The name of the annotation.
+     * @param class-string<T> $annotationName The name of the annotation.
      *
-     * @return object|null The Annotation or NULL, if the requested annotation does not exist.
+     * @return T|null The Annotation or NULL, if the requested annotation does not exist.
+     *
+     * @template T
```

Comment:
This is a very unusual place for `@template` :smile:

Reply:
Perhaps https://github.com/lcobucci/coding-standard/blob/7.0.x/src/Lcobucci/ruleset.xml#L17 can help?